File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    for(int i = 0 ; i < commands.size(); i++){
        vector<int>tmp;
        int start = commands[i][0];
        int end = commands[i][1];
        int k = commands[i][2];
        
        for(int j = start-1; j < end; j++){
            tmp.push_back(array[j]);
        }
        sort(tmp.begin(),tmp.end());
        answer.push_back(tmp[k-1]);
    }
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;
class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        int [] temp_arr = {};
        for(int i = 0 ; i < commands.length; i++){
            int start = commands[i][0];
            int end = commands[i][1];
            int k = commands[i][2];
            temp_arr = Arrays.copyOfRange(array, start-1, end);                   Arrays.sort(temp_arr);
            answer[i] = temp_arr[k-1];
        }

        return answer;
    }
}